### PR TITLE
#82 Resolve Bug: Duplicate observation for group USUBJID and VISIT

### PR DESCRIPTION
**Pull request by PoojaKumari05**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: data-raw/sv.R [hunk 1]
```diff
@@ -4,9 +4,23 @@
...
+    ASEQ = row_number(),
+    n = n()
+  ) %>%
+  mutate(
+    VISIT = ifelse(n > 1, paste0("UNSCHEDULED ", floor(VISITNUM), ".", ASEQ), VISIT),
+    VISITNUM = ifelse(n > 1, as.numeric(paste0(floor(VISITNUM), ".", ASEQ)), VISITNUM)
+  ) %>%
+  ungroup() %>%
+  select(-ASEQ, -n)
+
```
Comment (bundfussr) [inline data-raw/sv.R:23]: The labels of `VISIT` and `VISITNUM` got lost.
image

Could you add them?

File: R/data.R [hunk 2]
```diff
@@ -218,7 +218,8 @@
 
 #' Subject Visits Dataset
 #'
-#' A SDTM SV dataset from the CDISC pilot project
+#' A SDTM SV dataset from the CDISC pilot project. Duplicate observation for group
+#' variable USUBJID and VISIT is corrected.
```
Comment (bundfussr) [inline R/data.R:222]: Please use backticks for variable names, i.e., `USUBJID` instead of USUBJID.

File: inst/WORDLIST [hunk 3]
```diff
@@ -35,6 +35,7 @@ SUPPTR
 SV
 TAs
 Thampi
+USUBJID
```
Comment (bundfussr) [inline inst/WORDLIST:38]: This can be removed if backticks are used in `R/data.R`.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: inst/WORDLIST (hunk 3 above)
Comment (PoojaKumari05) [reply inst/WORDLIST]: Updated as suggested.

File: data-raw/sv.R (hunk 1 above)
Comment (PoojaKumari05) [reply data-raw/sv.R:29]: Updated the code to add labels.

File: R/data.R (hunk 2 above)
Comment (PoojaKumari05) [reply R/data.R]: Updated as suggested

File: inst/WORDLIST (hunk 3 above)
Comment (PoojaKumari05) [reply inst/WORDLIST]: @bundfussr - After removing the USUBJID from the wordlist the following check is failing. Please suggest.
image

Comment (bundfussr) [reply inst/WORDLIST]: @PoojaKumari05 , there was a line missing in the `DESCRIPTION` file. Therefore then man files were not created correctly. I have fixed it.

**Comment by PoojaKumari05** [comment]
Hi @bundfussr, the update in DESCRIPTION file is now throwing a warning message for the roxygen2 version - Is there a need to update the version in lockfile too OR having a warning is acceptable?
image

**Comment by bundfussr** [comment]
> Hi @bundfussr, the update in DESCRIPTION file is now throwing a warning message for the roxygen2 version - Is there a need to update the version in lockfile too OR having a warning is acceptable? image

You need to install the latest version of `roxygen2` in your RStudio Session. Then the warning should disappear.

In `admiral` the `renv.lock` file is no longer used. @kaz462 , should we remove `renv.lock` and `renv` in pharmaversesdtm?

**Review by bundfussr — APPROVED** [review]

**Review by kaz462 — APPROVED** [review]
@PoojaKumari05 @bundfussr Thank you both!

**Comment by kaz462** [comment]
> In `admiral` the `renv.lock` file is no longer used. @kaz462 , should we remove `renv.lock` and `renv` in pharmaversesdtm?

good point! issue created: https://github.com/pharmaverse/pharmaversesdtm/issues/89

**Comment by kaz462** [comment]
hi @manciniedoardo @bundfussr I'm not able to merge this PR, is this due to access?
image

**Comment by manciniedoardo** [comment]
> hi @manciniedoardo @bundfussr I'm not able to merge this PR, is this due to access? 

image
 
hmm I see the same... I think it's because of some of the checks not initiating at all - @dgrassellyb do you have any idea why this may be happening?

**Pull request merged**